test_rmw_implementation/test/test_publisher.cpp

```diff
@@ -375,3 +375,59 @@ TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), count_mismatched_subscri
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, subscription_count);
 }
+
+class CLASSNAME (TestPublisherUseLoan, RMW_IMPLEMENTATION)
+  : public CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION)
+{
+protected:
+  using Base = CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION);
+
+  void SetUp() override
+  {
+    Base::SetUp();
+    // Check if loaning is supported by the implementation
+    if (!pub->can_loan_messages) {
```

Comment:
@Lobotuerk oh, didn't know about this one! +1000

Reply:
I can't take the credit, @Blast545 told me about it